### ISSUE TITLE
Running knossos.exe in Windows VM

### ISSUE THREAD

**Issue by adwanner** [knossos-project/knossos#402]
I am trying to run knossos.exe (5.0) in a Windows 7 64bit VM on Ubuntu 14.04. 
However, KNOSSOS crashes while displaying the splace screen. If I run knossos.exe from cmd it outputs the following before crashing: 
```
Warning: Warning: Setting a new default format with a different version or profile after the global shared context is created may cause issues with context sharing.
Debug: 1.1.0 Microsoft Corporation GDI Generic
Debug: initializeOpenGLFunctions failed
```

**Comment by Optiligence** [comment]
Have you installed 3d drivers? KNOSSOS needs OpengGL 2.1 support.

**Comment by adwanner** [comment]
Thanks, that was indeed the issue. I forgot to switch on 3D acceleration support. (for your reference: https://www.howtogeek.com/124479/how-to-enable-3d-acceleration-and-use-windows-aero-in-virtualbox/)

**Comment by adwanner** [comment]
I installed the necessary GuestAddition and switched on 3D acceleration support in the VM (which is apparently still experimental, see https://www.virtualbox.org/manual/ch14.html and https://www.virtualbox.org/manual/ch04.html#ftn.idm2226 for more information).

I am now able to start and interact with KNOSSOS. However, there is an issue with the different viewports (see attached screenshot:  
![screenshot from 2017-05-05 15 13 44](https://cloud.githubusercontent.com/assets/17795672/25748792/acaae862-31ac-11e7-9ea0-0befee58220f.png) )

Basically, only one viewport is rendered properly, the others are either not visible, not rendered properly or just (although kind of functional) copies of the one that is properly rendered. Which viewport is rendered properly is kind of arbitrary. It can be any of YX, YZ or ZX. Depending on which one gets rendered properly, the 3D/skeleton viewport is rendered correctly or not.

If I click "reset VP position", Windows crashes.

Can you reproduce this or do you have any experience with other VMs that work properly?

**Comment by Optiligence** [comment]
Do other OpenGL apps work properly? 
There is #265 which is seems to be a very rare driver related issue. 
I know strange rendering bugs (opengl canvases rendering on top of everything else – even when the window was minimized – and moving slower than the rest of the window) from using KNOSSOS with 3D acceleration in a linux guest. 

I can attest that a linux vm with software rendering works (which i use for testing) but that’s probably not what you want. 
Your best bet with your current system was junest because it has acceptable performance. 
Other than that it’s probably wise to update your ubuntu installation, or – for the sake of it – just spin up a live system from a usb stick and use the AppImage.

**Comment by adwanner** [comment]
In general, OpenGL support is apparently experimental in virtual boxes. For various reasons it would be great to be able to run KNOSSOS on virtual (windows) machines, but so far I was not able to get the hardware acceleration working. 
The current solution is to switch to software open GL mode by copying opengl32.dll (available e.g. from http://download.blender.org/ftp/sergey/softwaregl/win64/opengl32.dll) next to knossos.exe.
Unfortunately opengl32.dll rendering is slow on VMs and Knossos is lagging, but otherwise it works so far smoothly. 
Anyway, it would be great to use hardware acceleration with Knosssos on VMs.

**Comment by jmrk84** [comment]
This really depends on the VM software used. My impression was that VMWare has better opengl support (I think VMWare player is also free), and Parallels for Mac works also from my experience.

**Comment by Optiligence** [comment]
Yeah, my arch linux vm also worked well with hw acc in vmware. 
With vbox i have to tinker the hw acc support myself because it isn’t enabled out of the box by the arch packages. And even then it doesn’t work that well, which is why i use software rendering instead. 

But i think it should work with a current Ubuntu vm or similar too. Not sure why it doesn’t work with win7.

**Comment by adwanner** [comment]
I just tested VMWare (Windows 10, host: Ubuntu 14.04, 3D acceleration enabled): Knossos runs with OpenGL hardware acceleration without crashing, but the viewports do not update, unless I minimize/maximize KNOSSOS.

Like with VirtualBox, rendering works in OpenGL software mode, but rendering is lagging.

**Comment by Optiligence** [comment]
I’m a bit reluctant to, but i wanna blame this on your host, same with the not-working junest. 
Can you switch between open source and closed source driver?

**Comment by Optiligence** [comment]
I tested a little: 
The KNOSSOS v5.0.1 AppImage works with 3d acc in VMWare on
- Arch Linux hosting Ubuntu 17.04
- Windows 10 hosting Arch Linux

Both combinations result in black viewports when started in VirtualBox with 3d acc for me. 

Since KNOSSOS 4.1.2 works (in the Arch guest), i filed a bug report https://bugreports.qt.io/browse/QTBUG-60671, we’ll see what comes from this.

**Comment by Optiligence** [comment]
note: test vbox with Qt 5.9
